### DISCUSSION TITLE
NullPointerException in io.apicurio.registry.serde.avro.AvroKafkaSerializer using Debezium Server

### DISCUSSION THREAD

**Discussion by koliankolin** [Apicurio/apicurio-registry#4155] [Q&A]
Hello!

I have this configuration for Debezium Server:
```
    DEBEZIUM_SINK_TYPE: kafka
    DEBEZIUM_SINK_KAFKA_PRODUCER_KEY_SERIALIZER: org.apache.kafka.common.serialization.StringSerializer

    DEBEZIUM_SINK_KAFKA_PRODUCER_VALUE_SERIALIZER: io.apicurio.registry.serde.avro.AvroKafkaSerializer
    DEBEZIUM_SINK_KAFKA_PRODUCER_APICURIO_REGISTRY_URL: http://apicurio:8080/apis/registry/v2
    DEBEZIUM_SINK_KAFKA_PRODUCER_APICURIO_REGISTRY_AUTO-REGISTER: true
    DEBEZIUM_SINK_KAFKA_PRODUCER_APICURIO_REGISTRY_FIND-LATEST: true
```

And I have this error:
```
"causedBy": {
          "exception": {
            "refId": 3,
            "exceptionType": "java.lang.NullPointerException",
            "message": null,
            "frames": [
              {
                "class": "io.apicurio.registry.resolver.config.DefaultSchemaResolverConfig",
                "method": "autoRegisterArtifact",
                "line": 89
              },
              {
                "class": "io.apicurio.registry.resolver.DefaultSchemaResolver",
                "method": "configure",
                "line": 67
              },
```

In source code this option should be added. I can't understand what I do wrong

**Comment by Ganesh-403** [comment]
@koliankolin The `NullPointerException` in:

```text
io.apicurio.registry.resolver.config.DefaultSchemaResolverConfig.autoRegisterArtifact()
```

suggests that the serializer is **not receiving the expected Apicurio configuration**, rather than there being a problem with `auto-register` itself.

The properties you've shown:

```yaml
DEBEZIUM_SINK_KAFKA_PRODUCER_APICURIO_REGISTRY_URL
DEBEZIUM_SINK_KAFKA_PRODUCER_APICURIO_REGISTRY_AUTO-REGISTER
DEBEZIUM_SINK_KAFKA_PRODUCER_APICURIO_REGISTRY_FIND-LATEST
```

look correct at first glance, but Debezium Server passes these through to the Kafka producer, and the Apicurio serializer expects the standard serializer properties such as:

```properties
apicurio.registry.url
apicurio.registry.auto-register
apicurio.registry.find-latest
```

If those properties are not propagated correctly, `DefaultSchemaResolverConfig` can end up with a missing (`null`) configuration value, which would explain the NPE. :contentReference[oaicite:0]{index=0}

One thing I'd also verify is the exact environment variable names. In YAML, keys containing hyphens (for example `AUTO-REGISTER`) can be interpreted differently depending on how they're translated into configuration properties. Most Debezium examples use the equivalent property names in connector configuration, for example:

```properties
apicurio.registry.url=http://apicurio:8080/apis/registry/v2
apicurio.registry.auto-register=true
apicurio.registry.find-latest=true
```

or, for Kafka Connect:

```properties
value.converter.apicurio.registry.auto-register=true
value.converter.apicurio.registry.find-latest=true
```

which are known to work. :contentReference[oaicite:1]{index=1}

A few things I'd check:

1. Which **Debezium Server version** are you using?
2. Which **Apicurio Registry client/serde version** is on the classpath?
3. Can you enable DEBUG logging and verify that the serializer actually sees:
   - `apicurio.registry.url`
   - `apicurio.registry.auto-register`
   - `apicurio.registry.find-latest`
4. If possible, could you share the full stack trace? The frame above `DefaultSchemaResolverConfig.autoRegisterArtifact()` often makes it easier to determine which configuration value is `null`.

From what I can tell, I couldn't find documentation indicating that `auto-register=true` by itself should cause a `NullPointerException`. If the properties are being passed correctly and you're still seeing the NPE, this looks more like a bug (or an incompatibility between the Debezium Server version and the Apicurio SerDe version) than an expected configuration issue.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
